packages/webpack/mixins/esbuild/mixin.core.js

```diff
@@ -0,0 +1,89 @@
+const { Mixin } = require('hops-mixin');
+const importComponentTransform = require.resolve(
+  './importComponentTransform.js'
+);
+
+const isLoaderFor = (extention) => (test) => {
+  if (Array.isArray(test)) {
+    return test.some(isLoaderFor(extention));
+  } else if (typeof test === 'object' && typeof test.test === 'function') {
+    return test.test(extention);
+  } else if (typeof test === 'function') {
+    return test(extention) === true;
+  } else if (typeof test === 'string') {
+    return test === extention;
```

Comment:
If the value of `test` is of type `String`, the condition is fulfilled when the filepath either starts with the `test`-string or equals the `test`-string. So I think you should write a warning in this case, that the test is not sufficient, and either do `test.endsWith(extension)` or simply return `false`.

Btw: what do you think of…?

```js
switch (true) {
  case Array.isArray(test):
    return test.some(isLoaderFor(extention));
  case typeof test === 'object' && typeof test.test === 'function':
    return test.test(extention);
  case typeof test === 'function':
    return test(extention) === true;
  case typeof test === 'string':
    return test.endsWith(extention);
  default:
    return false;
}
```